pkg/httpserver/httpserver.go

```diff
@@ -101,6 +101,8 @@ func (s *HTTPServer) Stop() error {
 	s.server = &http.Server{
 		Addr:    fmt.Sprintf(":%d", s.port),
 		Handler: s.httpServeMux,
+		// Needs a default for gosec. This can probably be brought down to a lower value.
+		ReadHeaderTimeout: time.Second * 10,
```

Comment:
#4913 from @jaellio is also open to address this by ignoring the error. In general I'd prefer to fix it like this does rather than ignore it, but I admit I don't have a good sense for exactly what we should set this to.

Reply:
There are multiple other places where we ignore G112. If we decide to make the fix, I think we should fix it in the other locations as well.

Reply:
done!